src/special/jest.js

```diff
@@ -0,0 +1,116 @@
+import path from 'path';
+import lodash from 'lodash';
+
+const _ = lodash;
+
+const jestConfigRegex = /jest.conf(ig|).js(on|)$/;
+const supportedProperties = [
+  'dependencyExtractor',
+  'preset',
+  'prettierPath',
+  'reporters',
+  'runner',
+  'setupFiles',
+  'setupFilesAfterEnv',
+  'snapshotResolver',
+  'snapshotSerializers',
+  'testEnvironment',
+  'testResultsProcessor',
+  'testRunner',
+  'transform',
+  'watchPlugins',
+];
+
+function parse(content) {
+  try {
+    return JSON.parse(content);
+  } catch (error) {
+    return {}; // ignore parse error silently
+  }
+}
+
+function contain(array, dep, prefix) {
+  if (!array) {
+    return false;
+  }
+
+  if (typeof array === 'string') {
+    return contain([array], dep, prefix);
+  }
+
+  // extract name if wrapping with options
+  const names = array.map(item => (lodash.isString(item) ? item : item[0]));
+  if (names.indexOf(dep) !== -1) {
+    return true;
+  }
+
+  if (prefix && dep.indexOf(prefix) === 0) {
+    return contain(array, dep.substring(prefix.length), false);
+  }
+
+  return false;
+}
+
+function removeNodeModuleRelativePaths(filepath) {
+  if (Array.isArray(filepath)) {
+    return removeNodeModuleRelativePaths(filepath[0]);
+  }
+  return filepath.replace(/^.*node_modules\//, '').replace(/\/.*/, '');
+}
+
+function filter(deps, options) {
+  const runner = deps.filter(dep => (
+    contain(options.runner, dep, 'jest-runner-')
+  ));
+
+  const watchPlugins = deps.filter(dep => (
+    contain(options.watchPlugins, dep, 'jest-watch-')
+  ));
+
+  const otherProps = lodash(options)
+    .entries()
+    .map(([prop, value]) => {
+      if (prop === 'transform') {
+        return _.values(value).map(removeNodeModuleRelativePaths);
+      }
+      if (Array.isArray(value)) {
+        return value.map(removeNodeModuleRelativePaths);
+      }
+      return removeNodeModuleRelativePaths(value);
+    })
+    .flatten()
+    .intersection(deps)
+    .value();
+
+  return _.uniq(runner.concat(watchPlugins).concat(otherProps));
+}
+
+function checkOptions(deps, options = {}) {
+  const pickedOptions = lodash(options)
+    .pick(supportedProperties)
+    .value();
+  return filter(deps, pickedOptions);
+}
+
+export default function parseJest(content, filePath, deps, rootDir) {
+  const filename = path.basename(filePath);
+  if (jestConfigRegex.test(filename)) {
+    try {
+      // eslint-disable-next-line global-require
+      const options = require(filePath) || {};
```

Comment:
I'm not sure it's safe to `require` a jest javascript configuration file. 

We have a util function `loadConfig` that tries to get json, yaml or js files by executing them in a sandbox.

It's a config file but we must be careful because the config files could have some side effects that could do unuseful stuff.

Reply:
I just copied what the `webpack.js` special handler does. I believe you have to require it locally because you can `require` inside of `jest.config.js`. The requirement is simply that it must evaluate to something that is JSON-serializable.

Reply:
Example of using [require in jest](https://jestjs.io/docs/en/configuration#defaults_

Reply:
@rumpl after switching from `require` to `evaluate` for javascript, I get 1 failing test and it's for the example above where `require` is used. There's an error thrown for
```
ReferenceError: require is not defined
    at evalmachine.<anonymous>:1:1
    at Script.runInContext (vm.js:102:20)
    at Script.runInNewContext (vm.js:108:17)
    at Object.runInNewContext (vm.js:291:38)
```